### PR TITLE
Debug volunteer classifications by subjectID

### PR DESCRIPTION
**Pull request by eatyourgreens**
Add a form to the user admin page, which filters their classifications by classified subject ID. Handy if you want to debug reports of duplicate subjects for a volunteer.

Screenshot of the user account admin form, with a text input that can be used to filter the classifications history by subject ID.

Staging branch URL: https://pr-6426.pfe-preview.zooniverse.org

# Required Manual Testing

- [ ] Does the non-logged in home page render correctly?
- [ ] Does the logged in home page render correctly?
- [ ] Does the projects page render correctly?
- [ ] Can you load project home pages?
- [ ] Can you load the classification page?
- [ ] Can you submit a classification?
- [ ] Does talk load correctly?
- [ ] Can you post a talk comment?

# Review Checklist

- [ ] Does it work in all major browsers: Firefox, Chrome, Edge, Safari?
- [ ] Does it work on mobile?
- [ ] Can you `npm ci` and app works as expected?
- [ ] If the component is in coffeescript, is it converted to ES6? Is it free of eslint errors? Is the conversion its own commit?
- [ ] Are the tests passing locally and on GitHub Actions?

# Optional

- [ ] Have you replaced any `ChangeListener` or `PromiseRenderer` components with code that updates component state?
- [ ] If changes are made to the classifier, does the dev classifier still work?
- [ ] Have you [resized and compressed](https://developers.google.com/web/fundamentals/performance/optimizing-content-efficiency/image-optimization) any image you've added?
- [ ] Have you added in [flow type annotations](https://flowtype.org/docs/type-annotations.html)?
- [ ] Have you followed the [Springer guidelines for commit messages](https://github.com/springernature/frontend-playbook/blob/master/git/git.md#commit-messages)?

### CODE REVIEW

File: app/pages/admin/user-settings.jsx [hunk 1]
```diff
@@ -104,6 +113,12 @@ class UserSettings extends Component {
         </details>
         <details>
           <summary>Recent classifications {this.state.classifications.length}</summary>
+          <form onSubmit={this.updateSubjectID}>
+            <label for="subjectId">
+              Filter by subject ID:
+              <input id="subjectID" name="subjectID" type="text" defaultValue='' />
```
Comment (eatyourgreens) [inline app/pages/admin/user-settings.jsx:119]:
```suggestion
              <input id="subjectId" name="subjectID" type="text" defaultValue='' />
```

**Review by shaunanoordin — APPROVED** [review]
## PR Review

Affects: Admin -> Manage User

This PR allows admins - when viewing the Classifications of a target user in the Admin section - to specify which Subject ID they want to see. This results in only Classifications _from_ the target user _for_ that specific Subject to be shown.

- New input box is located under [(target user's page)](https://pr-6426.pfe-preview.zooniverse.org/admin/users/1459668?env=production) -> Classification History -> Recent Classifications
  - This ONLY shows if we've enabled _admin mode._
  - Input box accepts any Subject ID, (all other forms of input will be safe but effectively ignored by the API,) and will refresh the "Recent Classifications" list. (Presumably only up to 20, if the target user somehow managed to submit more than 20 classifications for a given Subject. Maybe they really Fancy Cats.)
  - *️⃣ The input requires user to press the Enter key (i.e. submit the form) to trigger an update.

Code read looks good and functionality checks out.

### Dev Notes

[UI/UX][minor] One small UI comment is that the "press Enter to trigger an update" *️⃣ action doesn't conform to the UI behaviour that's so common to the rest of the admin page (i.e. commit update _on input change_ )

A small recommendation is to either (XOR):
- Trigger updateSubjectID on the input's onChange event (keeping in line with the rest of PFE admin's behaviour), or
- Add a small "submit" button to make it clear to admin that they need to manually trigger the form submit/update action.

### Testing

Tested on macOS + Chrome, logged in as admin, viewing user `zootester1`, using Subject ID 41843263

Test URL: https://pr-6426.pfe-preview.zooniverse.org/admin/users/1459668?env=production
Testing steps:
- Login as admin user, view test URL, enable admin mode
- Type in a Subject ID that zootester1 has recently classified, press enter. Confirm that Recent Classifications list list only Clasifications for that Subject.
- Type in a random Subject ID, press enter. Confirm that Recent Classifications lists nothing.

### Status

LGTM 👍

**Comment by eatyourgreens** [comment]
Good point about the submit button. That’s needed to get this working on a phone too.

**Comment by eatyourgreens** [comment]
I’m not sure about using the `change` event. You’d have to Tab out of the text box in order to fire the `change` handler. That doesn’t seem obvious to me. It’s definitely annoying in the project builder, where changes aren’t saved until a text box loses focus.

**Pull request merged**